client/my-sites/checkout/composite-checkout/components/checkout-main.tsx

```diff
@@ -338,9 +343,14 @@ export default function CheckoutMain( {
 		// Only wait for stored cards to load if we are using cards
 		( allowedPaymentMethods.includes( 'card' ) && isLoadingStoredCards );
 
-	const contactDetails = useSelect( ( select ) => select( 'wpcom-checkout' )?.getContactInfo() );
-	const recaptchaClientId = useSelect( ( select ) =>
-		select( 'wpcom-checkout' )?.getRecaptchaClientId()
+	const contactDetails: ManagedContactDetails = useSelect(
+		( select ) => ( select( 'wpcom-checkout' ) as WpcomCheckoutStoreSelectors )?.getContactInfo(),
+		[]
+	);
+	const recaptchaClientId: number = useSelect(
```

Comment:
This too could be undefined. Why do we need to add an explicit type? Is there a problem with the inference after this upgrade?